packages/medusa/src/api-v2/admin/users/route.ts

```diff
@@ -43,16 +50,49 @@ export const POST = async (
   req: AuthenticatedMedusaRequest<CreateUserDTO>,
   res: MedusaResponse
 ) => {
-  const workflow = createUsersWorkflow(req.scope)
+  const userModuleService = req.scope.resolve<IUserModuleService>(
+    ModuleRegistrationName.USER
+  )
+  const authModuleService = req.scope.resolve<IAuthModuleService>(
+    ModuleRegistrationName.AUTH
+  )
+
+  const { jwt_secret } = req.scope.resolve("configModule").projectConfig
+
+  let authUser = await authModuleService.retrieve(req.auth.auth_user_id)
+  let token = jwt.sign(authUser, jwt_secret)
+
+  let user: UserDTO | null = null
+
+  if (req.auth.actor_id) {
```

Comment:
I am a bit on the fence about how we should handle this scenario.

If an `actor_id` is assigned to the `req.auth` object, it means the request carries authentication in the form of either a bearer or a session token, AND that this authenticated already has an associated user. 

I believe, there are two "sub-scenarios" that can take place here:
1. Throw error if the email provided doesn't match the email of the authenticated user (current implementation)
2. Respond with the already authenticated user, even though the email might be different

I chose 1. as I think it would be unintuitive to respond with a user with a different email from what was passed in a request to create a new user. 

I am curios to hear what others think.

Reply:
If we proceed with 1., I will need to update a few modules tests, that are currently failing e.g. `create-user.spec.js` (this test illustrates my point well, as it is failing right now because of an email mismatch).

Reply:
This is the `POST` endpoint to create a new user. Could you have missed this, or does your question still stand in this context?

Reply:
Agh the diff showed GET above your changes, and I didn't check the line numbers, my bad.

So the POST is only to get a new token, but you cannot update info on the user itself (eg. first name)? If it is just for renewing a token, then my comment still stands, there is no need to pass an email IMO as it doesn't provide any added security.

Reply:
You can indeed pass first name and last name in the request.

The flow to create a new authenticated user is:
1. Create an AuthUser (`@medusajs/auth`)
``` 
POST /auth/admin/emailpass
{ email: "test@email.com", password: "supersecret" }
```

This will return a token signed with the AuthUser. The idea is that this token carries the authentication for the subsequent request to create an actual User.

2. Create a User (`@medusajs/auth`)
```
POST /admin/users
{ email: "test@email.com", first_name: "John", last_name: "Doe" }
{ Authorization: "Bearer [token-from-first-request]" }
```

On the server, we then perform the following actions as part of 2.:
1. In the `authenticate-middleware.ts`, we extract the AuthUser from the bearer token. This is passed along as `req.auth`.
2. In the API route creating a user, we then extract the `auth_user_id` from `req.auth` and retrieve the AuthUser
3. Then, we extract the `actor_id` from `req.auth`. 
4. If `actor_id` is present, we retrieve the User
5. else, we create a new User and associate it with the AuthUser using the `auth_user_id` extracted from `req.auth`
6. Finally, we respond with the created User and a fresh token signed with the AuthUser

My question relates to bullet 4. If `actor_id` is present in `req.auth`, it means the AuthUser attached to the request has been used previously to create a User. For example, this could happen if you cURL `POST /admin/users` with the bearer token as authorization header multiple times.

Now, in case you pass a different email, do we:
1. Respond with the User, even though the email of that User does not match the one passed in the request
2. Throw

Reply:
Discussed in slack, and I understood the background behind this better. We also concluded it's better to throw if you try to associate an already associated AuthUser.